src/main/main.ts

```diff
@@ -271,25 +281,36 @@ app.on('ready', async () => {
 
   await createWindow()
   log('created windows')
-  mainWindow.webContents.on('did-fail-load', () => {
-    log('failed loading')
-  })
+  if (isBrowserWindow(mainWindow)) {
```

Comment:
instead of wrapping each interaction with `mainWindow` in `if (isBrowserWindow)...` let's do it once and be over with it

```
if (!isBrowserWindow(mainWindow)) {
  throw new Error(`mainWindow is on unexpected type {mainWindow}`)
}
```

this way everything in this function below this line will already have proper type asserted by the guard